frontend/src/components/modals/ChangePasswordModal.tsx

```diff
@@ -11,29 +12,35 @@ import { ModalHeader } from './modalparts/ModalHeader';
 import { errorState, Input } from '../forms/FormField';
 import { paths } from '../../routers/paths';
 import '../../shared.scss';
+import css from './ChangePasswordModal.module.scss';
 
 import { LoadingSpinner } from '../LoadingSpinner';
-
 import { StoreDispatchType } from '../../redux';
 import { userActions } from '../../redux/user';
 
-export const ConfirmPasswordModal: Modal<ModalTypes.CONFIRM_PASSWORD_MODAL> = ({
+const classes = classNames.bind(css);
+
+export const ChangePasswordModal: Modal<ModalTypes.CHANGE_PASSWORD_MODAL> = ({
   closeModal,
   actionData,
   deleteUser,
 }) => {
   const history = useHistory();
   const { t } = useTranslation();
   const dispatch = useDispatch<StoreDispatchType>();
-  const [password, setPassword] = useState('');
-  const [errorMessage, setErrorMessage] = useState('');
+  const [currentPassword, setCurrentPassword] = useState('');
+  const [newPassword, setNewPassword] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
   const [isLoading, setIsLoading] = useState(false);
   const formRef = useRef<HTMLFormElement>(null);
 
+  const [errorMessage, setErrorMessage] = useState('');
+  const confirmationError = errorState(useState(''));
+
   const errorHandler = (response?: { data: string; status: number }) => {
     switch (response?.status) {
       case 400: {
-        setErrorMessage(t('Bad request error - is the password correct?'));
+        setErrorMessage(t('Bad request error - is current password correct?'));
```

Comment:
This didn't have a translation to start with, but it would be good to add while changing the key here

Reply:
Translation added.